lib/oli/resources.ex

```diff
@@ -339,6 +339,7 @@ defmodule Oli.Resources do
           time_limit: previous_revision.time_limit,
           scope: previous_revision.scope,
           retake_mode: previous_revision.retake_mode,
+          assessment_mode: previous_revision.assessment_mode,
```

Comment:
Nice catch remembering to update this function. This merging of attrs really exploded and been a previous source of bugs, I wonder if we should take the opportunity to change the approach here and instead of explicitly specifying each attr we want to carry forward, maybe we should just merge all attrs from `previous_revision` and only drop any that we don't want to carry forward (which currently, I think is none since we are merging all of them). Thoughts?

Reply:
Makes sense. I will apply the change